### DISCUSSION TITLE
How come the session flow isn't used in this SDK?

### DISCUSSION THREAD

**Discussion by chrispaynter** [Adyen/adyen-react-native#327] [General]
We have built an integration with Web using the standard session flow, where we create the session on the backend, and then pass into the drop in and await a webhook call.

We would now like to follow the same flow, but we see it's not implemented in this RN SDK.

Is this something that's coming soon?

*UPDATE*

I've started prodding around at how we could potentially add this.

This code is basically pseudo, but in the case of iOS would it be a case of simply adding a new method to open the drop in based on session?

Something like this?

@descorp I thought I'd tag you here as I saw in the commit history you wrote the original `open` method in [that file](https://github.com/Adyen/adyen-react-native/blob/ac33a1a7ddbbce313d619ac37edbea3ef6cfd8c5/ios/Components/DropInModule.swift#L27).

```
@objc
    func openWithSession(_ sessionToken: String, initialData: NSDictionary, configuration: NSDictionary) {
        let parser = RootConfigurationParser(configuration: configuration)
        let clientKey: String
        do {
            clientKey = try fetchClientKey(from: parser)
        } catch {
            return sendEvent(error: error)
        }

         guard let apiContext = try? APIContext(environment: parser.environment, clientKey: clientKey) else { return }

        let context: AdyenContext
        if let payment = parser.payment {
            (try? ApplepayConfigurationParser(configuration: configuration).buildConfiguration(payment: payment)).map {
                config.applePay = $0
            }

            // TODO: add analyticsConfiguration: AnalyticsConfiguration()
            context = AdyenContext(apiContext: apiContext, payment: payment)
            
            
        } else {
            context = AdyenContext(apiContext: apiContext, payment: nil, analyticsConfiguration: AnalyticsConfiguration())
        }

        let configuration = AdyenSession.Configuration(sessionIdentifier: response.sessionId,
                                               initialSessionData: response.sessionData,
                                               context: context)

        AdyenSession.initialize(with: configuration, delegate: self, presentationDelegate: self) { [weak self] result in
          switch result {
          case let .success(session):
              // store the session object
              self?.session = session

              // Now display the component driven by session data
              let component = DropInComponent(paymentMethods: session.sessionContext.paymentMethods,
                                            context: context,
                                            configuration: config)

              let config = DropInConfigurationParser(configuration: configuration).configuration
              currentComponent = component
              component.delegate = self
              present(component: component)
          case let .failure(error):
              // handle the error
              throw error("not implemented yet")
          }
        }

    }
```

**Comment by descorp** [comment]
Hey @chrispaynter 

"standard session flow" introduced on mobile with v5.x.x of iOS and Android SDKs.
Current stable version of React Native still points to v4.x.x versions.

We have already adopted v5.x.x(s) on `develop` and currently working on providing "sessions flow" support.
Our goal is to release `v2.0.0-aplha` in the next couple of weeks.
Stay tune!

**Comment by chrispaynter** [comment]
Thanks @descorp that's very promising to hear.

I guess in the mean time we'll need to implement a proxy backend for RN 😞 

Which branch do you have that code on?

**Reply by descorp** [reply]
Hey @chrispaynter 

It is in active development 😅
Some breaking changes may still be introduced.

If applicable, I would advice waiting for an ALPHA release in Jan 2024.

**Reply by chrispaynter** [reply]
Hi @descorp , thanks for this.

How are you tracking for the alpha this month?

**Reply by descorp** [reply]
Hey @chrispaynter 

We haven't published it yet as ALPHA, but all the code is ready for testing in `develop`. 
Feel free to give it a try by cloning as "local dependency" - make sure to run `npm/yarn install` inside the `node_modules/@adyen/react-native` folder, GIT source is not pre-build by default.

Will keep you posted

**Reply by chrispaynter** [reply]
Thanks @descorp.

We ended up creating those proxy endpoints to get us launched in the meantime.

Any chance you could add documentation for utilising the session code (or link if it already exists)? Then I could resource some time for @anthlasserre to have a look.

We could probably help do some external testing here to fast track getting this out.

**Reply by descorp** [reply]
Thanks @chrispaynter 

So far we only have [example code](https://github.com/Adyen/adyen-react-native/blob/develop/example/src/Views/Checkout/SessionsCheckout.js)

Simply pass `/sessions` API response to `AdyenCheckout`:
```js
        <AdyenCheckout
          config={ ... }
          session={session}
          onComplete={didComplete}
          onError={didFail}
        >
          {/* Your Checkout view */}
        </AdyenCheckout>
```

**Reply by chrispaynter** [reply]
Thanks @descorp.

We might have a little bit of time up our sleeve to update to this.

How are you guys going with testing and getting the alpha out?

**Reply by descorp** [reply]
Hey @chrispaynter 

It is available as [v.2.0.0 Beta](https://www.npmjs.com/package/@adyen/react-native) feel free to try it out

**Reply by chrispaynter** [reply]
Nice one @descorp , thanks a lot.
